python/src/iceberg/utils/schema_conversion.py

```diff
@@ -147,7 +145,14 @@ def _resolve_union(self, type_union: dict | list | str) -> tuple[str | dict[str,
         else:
             avro_types = type_union
 
-        is_optional = "null" in avro_types
+        # For the Iceberg spec it is required to set the default value to null
+        # From https://iceberg.apache.org/spec/#avro
+        # Optional fields must always set the Avro field default value to null.
+        #
+        # This means that null has to come first:
+        # https://avro.apache.org/docs/current/spec.html
+        # type of the default value must match the first element of the union.
+        is_optional = "null" == avro_types[0]
```

Comment:
I'd prefer to be a bit more careful and work with types where `null` isn't first. I know it's required, but we should be strict with what this library produces and liberal with what it accepts.

Reply:
Nevermind, based on the comment above. I understand the limitation now.